fast64_internal/sm64/sm64_objects.py

```diff
@@ -132,8 +134,46 @@
 	('Toxic Haze', 'Toxic Haze', 'Toxic Haze')
 ]
 
+class InlineGeolayoutObjConfig:
+	def __init__(
+		self, name, geo_node,
+		can_have_dl=False,
+		must_have_dl=False,
+		must_have_geo=False,
+		uses_location=False,
+		uses_rotation=False,
+		uses_scale=False
+	):
+		self.name = name
+		self.geo_node = geo_node
+		self.can_have_dl = can_have_dl or must_have_dl
+		self.must_have_dl = must_have_dl
+		self.must_have_geo = must_have_geo
+		self.uses_location = uses_location
+		self.uses_rotation = uses_rotation
+		self.uses_scale = uses_scale
+
+inlineGeoLayoutObjects = {
+	'Geo ASM'				: InlineGeolayoutObjConfig('Geo ASM', 				FunctionNode),
+	'Geo Branch'			: InlineGeolayoutObjConfig('Geo Branch', 			JumpNode,
+								must_have_geo=True),
+	'Geo Translate/Rotate'	: InlineGeolayoutObjConfig('Geo Translate/Rotate', 	TranslateRotateNode,
+								can_have_dl=True, uses_location=True, uses_rotation=True),
+	'Geo Translate Node'	: InlineGeolayoutObjConfig('Geo Translate Node', 	TranslateNode,
+								can_have_dl=True, uses_location=True),
+	'Geo Rotation Node'		: InlineGeolayoutObjConfig('Geo Rotation Node', 	RotateNode,
```

Comment:
indentation (purely visual) looks messed up in `inlineGeoLayoutObjects` but maybe it's just github

Reply:
wow yeah github DESTROYED the formatting.. should i align with spaces instead? my local tab size is 4 and i'm used to using spaces as tabs

![image](https://user-images.githubusercontent.com/79979276/132068081-f4c0c169-6cf0-4a99-baea-8f03ea16f173.png)

Reply:
PEP-8 recommends indenting with 4-spaces instead of tabs and I'm definitely on that ship since various text editors can't handle tabs consistently... I'd rather have all indentation changed in a cleanup PR though so if it looks fine on your end then good

Reply:
awesome, yeah it'd be good to enable PEP-8 auto formatting sometime soon